sdk/core/azure-core/test/e2e/azure_core_storage_list_containers_sample.cpp

```diff
@@ -18,12 +18,13 @@ int main()
 {
 
   // Create the Transport
-  std::shared_ptr<HttpTransport> transport = std::make_unique<CurlTransport>();
+  Azure::Core::Http::TransportPolicyOptions transportOptions;
+  transportOptions.Transport = std::move(std::make_unique<CurlTransport>());
```

Comment:
I still find this an odd trade-off to make to hurt usability given the transport isn't optional. When you add the transport specific options too, like `CurlTransportOptions`, this because even more lines of code, just for setup.

Reply:
The transport adapter is optional to be specified by the end user.
Azure Core will provide a default transport adapter (libcurl for now) when it is not specified

Reply:
cc @JeffreyRichter

Reply:
This is what users would write if they want to pick the non-default transport:
```C++
    auto transportAdapter = std::make_shared<Azure::Core::Http::CurlTransport>(curlOptions);
    Azure::Core::Http::TransportPolicyOptions options;
    options.Transport = transportAdapter;
    auto transportPolicy = std::make_unique<Azure::Core::Http::TransportPolicy>(options);
```

Can we do this instead?
```C++
    auto transportAdapter = std::make_shared<Azure::Core::Http::CurlTransport>(curlOptions);
    auto transportPolicy = std::make_unique<Azure::Core::Http::TransportPolicy>(transportAdapter);
```

Reply:
Azure::Storage::XxxClientOptions options;
options.Transport.Adapter = std::make_shared<Azure::Core::Http::CurlTransport>(curlOptions);
auto client = BlobClient("url", cred, options);

Reply:
That makes sense and works well @JeffreyRichter, when the customer is initializing the client tor using client options.

What about when they want to override the transport policy?

**The question is:** Should `TransportPolicy` take `HttpTransport` as a required ctor argument with an optional `TransportPolicyOptions` argument?

Reply:
No. The TransportPolicy (like ALL other policies except telemetry) should take just the TransportOptions parameter (because that is where the HttpTransportAdaptor field is defined. We wouldn't want it in TransportOptions AND as a ctor parameter.